zookeeper-server/src/main/java/org/apache/zookeeper/KeeperException.java

```diff
@@ -404,7 +406,10 @@ public void setCode(int code) {
         /** The session has been closed by server because server requires client to do SASL authentication,
          *  but client is not configured with SASL authentication or configuted with SASL but failed
          *  (i.e. wrong credential used.). */
-        SESSIONCLOSEDREQUIRESASLAUTH(-124);
+        SESSIONCLOSEDREQUIRESASLAUTH(-124),
+        /** Operation was throttled and not executed at all. please, retry! We also use this
+         *  to signal exceeding number of ephemeral nodes per session. */
+        THROTTLEDOP (-127);
 
```

Comment:
- Why not choose the sequential number: `-125`?
- the client gets `KeeperErrorCode = Op throttled due to high load for /xxxx`. a little ambiguous
- `SessionEphemeralCountExceed(-125)` is better?

Reply:
Our internal version of ZK actually has more error codes here after `  SESSIONCLOSEDREQUIRESASLAUTH(-124)`, so to avoid backward compatibility issues in the future, I have just set the error code to match what it is with our internal verison. When we catch up on our upstreaming, it will be less messy.

Reply:
In response to your second point, we throw this for when our client exceed certain quota limits that we have set up. It is meant to be a very general error.